honeybadgermpc/mpc.py

```diff
@@ -85,26 +85,33 @@ def __init__(self, sid, n, t, myid, pid, send, recv, prog, config, **prog_args):
         share_buffer = [self._share_buffers[i][shareid] for i in range(self.N)]
 
         point = EvalPoint(self.field, self.N, use_fft=False)
+        # Maximum number of errors to tolerate. This needs to be
+        # `t` to ensure that we are in the robust setting.
+        c = self.t
         opening = robust_reconstruct(
-            share_buffer, self.field, self.N, t, point)
+            share_buffer, self.field, self.N, degree, c,
+            point, wait_for_all=wait_for_all)
         self._openings[shareid] = opening
         p, _ = await opening
         return p(self.field(0))
 
-    def open_share_array(self, sharearray):
+    def open_share_array(self, sharearray, wait_for_all=False):
         # Choose the shareid based on the order this is called
         shareid = len(self._openings)
+        degree = sharearray.t
 
         def _send(j, o):
             (tag, share) = o
             self.send(j, (tag, shareid, share))
         _recv = self._sharearray_buffers[shareid].get
-
+        # Maximum number of errors to tolerate. This needs to be
+        # `t` to ensure that we are in the robust setting.
+        c = self.t
```

Comment:
This would make the reconstruction wait for at least **degree+t+1** points. This corresponds to the maximum number of errors that can be tolerated.

If I understand you correctly, the `f` that we discussed is just a way to send faulty values and is completely independent of how you want to do batch reconstruction since in order to be robust you'd want to wait for at least **degree+t+1** points.